lib/bcu/options.rb

```diff
@@ -126,4 +118,78 @@ def self.validate_options(options)
       exit 1
     end
   end
+
+  def self.build_config(use_config_file: true)
+    options = use_config_file ? load_default_options : create_default_options
+
+    options.casks = nil
+    options.install_options = ""
+    options.list_pinned = false
+    options.pin = nil
+    options.unpin = nil
+    options.command = "run"
+
+    options
+  end
+
+  def self.load_default_options
+    config_filename = "#{ENV["HOME"]}/.brew-cu"
+    unless File.exist?(config_filename)
+      odebug "Config file doesn't exist, creating"
+      create_default_config_file config_filename
+    end
+
+    default_options = create_default_options
+    if File.exist?(config_filename)
+      odebug "Loading configuration from config file"
+      options = {}
+      File.open(config_filename) do |f|
+        options = (YAML.safe_load f.read).to_h
+        odebug "Configuration loaded", options
+      end
+      OpenStruct.new options
+    else
+      # something went wrong while reading config file
+      odebug "Config file wasn't created, setting default config"
+      default_options
+    end
+  end
+
+  def self.create_default_options
+    default_values = default_config_hash
+    default_options = OpenStruct.new
+    default_options.all = default_values["all"]
+    default_options.force = default_values["force"]
+    default_options.cleanup = default_values["cleanup"]
+    default_options.force_yes = default_values["force_yes"]
+    default_options.no_brew_update = default_values["no_brew_update"]
+    default_options.quiet = default_values["quiet"]
+    default_options.verbose = default_values["verbose"]
+    default_options.interactive = default_values["interactive"]
+    default_options.debug = default_values["debug"]
```

Comment:
the value `debug` is not in `default_values`

Reply:
should a loop over the hash and a `default_options[option_name] = default_values[option_name]` a little shorter and include all options from hash?

Reply:
i see there you also can write when you don't need to exclude some keys:
```ruby
default_options = OpenStruct.new(default_values)
```